src/maple_pool/depository.ts

```diff
@@ -0,0 +1,225 @@
+    public readonly collateralMint: PublicKey,
+    public readonly collateralDecimals: number,
+    public readonly collateralSymbol: string,
+    public readonly depositoryCollateral: PublicKey,
+    public readonly maplePool: PublicKey,
+    public readonly maplePoolLocker: PublicKey,
+    public readonly mapleGlobals: PublicKey,
+    public readonly mapleLender: PublicKey,
+    public readonly mapleSharesMint: PublicKey,
+    public readonly mapleLockedShares: PublicKey,
+    public readonly mapleLenderShares: PublicKey,
+    public readonly syrupProgramId: PublicKey
+  ) {}
+
+  public static async initialize({
+    connection,
+    uxdProgramId,
+    syrupProgramId,
+    collateralMint,
+    collateralSymbol,
+    maplePool,
+    maplePoolLocker,
+    mapleGlobals,
+    mapleSharesMint,
+  }: {
+    connection: Connection;
+    uxdProgramId: PublicKey;
+    syrupProgramId: PublicKey;
+    collateralMint: PublicKey;
+    collateralSymbol: string;
+    maplePool: PublicKey;
+    maplePoolLocker: PublicKey;
+    mapleGlobals: PublicKey;
+    mapleSharesMint: PublicKey;
+    cluster: Cluster;
+  }): Promise<MaplePoolDepository> {
+    // The depository is PDA from the pool and the collateral
+    const [depository] = PublicKey.findProgramAddressSync(
+      [
+        Buffer.from(MAPLE_POOL_DEPOSITORY_NAMESPACE),
+        maplePool.toBuffer(),
+        collateralMint.toBuffer(),
+      ],
+      uxdProgramId
+    );
+
+    // Read collateral decimals
+    const collateralToken = new Token(
+      connection,
+      collateralMint,
+      TOKEN_PROGRAM_ID,
+      null as unknown as Signer
+    );
+    const collateralInfo = await collateralToken.getMintInfo();
+    if (!collateralInfo) {
+      throw new Error('Cannot find the collateral mint');
+    }
+    const collateralDecimals = collateralInfo.decimals;
+
+    const depositoryCollateral = await this.findDepositoryCollateralAddress(
+      depository,
+      uxdProgramId
+    );
+
+    const mapleLender = await this.findLenderAddress(
+      maplePool,
+      depository,
+      syrupProgramId
+    );
+
+    const mapleLockedShares = await this.findLockedSharesAddress(
+      mapleLender,
+      syrupProgramId
+    );
+    const mapleLenderShares = await this.findLenderSharesAddress(
+      depository,
+      mapleSharesMint
+    );
+
+    return new MaplePoolDepository(
+      depository,
+      collateralMint,
+      collateralDecimals,
+      collateralSymbol,
+      depositoryCollateral,
+      maplePool,
+      maplePoolLocker,
+      mapleGlobals,
+      mapleLender,
+      mapleSharesMint,
+      mapleLockedShares,
+      mapleLenderShares,
+      syrupProgramId
+    );
+  }
+
+  private static async findDepositoryCollateralAddress(
+    depository: PublicKey,
+    uxdProgramId: PublicKey
+  ) {
+    return (
+      await PublicKey.findProgramAddress(
+        [
+          Buffer.from(MAPLE_POOL_DEPOSITORY_COLLATERAL_NAMESPACE),
+          depository.toBytes(),
```

Comment:
We usually use `toBuffer()`:

```suggestion
          depository.toBuffer(),
```

No diff between `toBuffer` and `toBytes`.

```
    /**
     * Return the byte array representation of the public key
     */
    toBytes(): Uint8Array;
    /**
     * Return the Buffer representation of the public key
     */
    toBuffer(): Buffer;
```

```
    static findProgramAddressSync(
      seeds: Array<Buffer | Uint8Array>,
      programId: PublicKey,
    ): [PublicKey, number];
```